File: src/main/java/com/github/tomasmilata/intelliroutes/RoutesFileTypeFactory.java
package com.github.tomasmilata.intelliroutes;

import com.intellij.openapi.fileTypes.*;
import org.jetbrains.annotations.NotNull;

public class RoutesFileTypeFactory extends FileTypeFactory {
  @Override
  public void createFileTypes(@NotNull FileTypeConsumer fileTypeConsumer) {
    fileTypeConsumer.consume(RoutesFileType.INSTANCE, "simple");
  }
}

File: src/main/java/com/github/tomasmilata/intelliroutes/RoutesFileTypeFactory.kt
package com.github.tomasmilata.intelliroutes

import com.intellij.openapi.fileTypes.*

class RoutesFileTypeFactory : FileTypeFactory() {
    override fun createFileTypes(fileTypeConsumer: FileTypeConsumer) {
        fileTypeConsumer.consume(RoutesFileType.INSTANCE, ExactFileNameMatcher("routes"))
    }
}